src/auframe/auframe.c

```diff
@@ -100,3 +105,41 @@ double auframe_level(struct auframe *af)
 
 	return af->level;
 }
+
+
+/**
+ * Computes audio frame volume and compares with a boundary to decide if it is
+ * nearly silence
+ *
+ * Note: faster than auframe_level for s16le
+ *   - no sqrt
+ *   - no logarithm
+ *   - early loop exit for non-silence
```

Comment:
I wonder if performance is really a problem? In my quick measurements its nearly the same (and we can reuse the level for `vumeter` etc.). Silence tracks are more expensive since comparison costs much more than addition.

Reply:
We do not need the vumeter on our products and the silence detection works already. We don't want to have 640 sqrt and logarithm operations for each frame.

Reply:
The sqrt is only done once per frame sum. not per sample.

Reply:
Okay, silence detection is only needed AJB_LOW and AJB_HIGH situations. But it is not the same like computing a volume in `dB`. This might justify a separate function `auframe_silence()`. The note can be removed, it is meant only for the review.

Reply:
> 

Ah, yes, you're right! Logarithm also. Thus there are only the `O(n)`
- `double` multiplications instead of `int`
- `double` additions instead of `int`

On one product we have a very old hardware (balckfin CPU). I will check how much the impact would be to replace the silence detection with the `auframe_level()` computation.

Reply:
Yes the double multiplications has much more impact, i will prepare a patch (i dont think we need this), here are some results:
````
auframe_silence 3468 usecs (2000)
auframe_silence2 665 usecs (2000)
auframe_silence 3572 usecs (2000)
auframe_silence2 645 usecs (2000)
```
https://gist.github.com/sreimers/67440f237cae34537c9c1667900a7c57



> On one product we have a very old hardware (balckfin CPU).

We can use maybe a square root alternative:

```c
static float square_root(float x)
{
	unsigned int i = *(unsigned int *)&x;

	// adjust bias
	i += 127 << 23;
	// approximation of square root
	i >>= 1;

	return *(float *)&i;
}
```

On modern hardware I see no difference with `sqrt`.

Reply:
Thanks, improvement here would be welcome! If the performance loss is only little, than I would remove the extra silence detection.

Reply:
@sreimers Do you have a sample program for this benchmarks?

Reply:
https://gist.github.com/sreimers/67440f237cae34537c9c1667900a7c57

And this patch: https://github.com/baresip/rem/pull/38

Reply:
thanks!